conans/test/functional/toolchains/gnu/test_autotools.py

```diff
@@ -0,0 +1,206 @@
+    os.rename(os.path.join(client.current_folder, "test_package"),
+              os.path.join(client.current_folder, "test_package2"))
+    client.save({"profile": profile})
+    client.run("create . --profile=profile")
+    print(client.out)
+
+    main = gen_function_cpp(name="main", includes=["hello"], calls=["hello"])
+    makefile = textwrap.dedent("""\
+        app: main.o
+        	$(CXX) $(CFLAGS) $(LDFLAGS) -o app main.o $(LIBS)
+
+        main.o: main.cpp
+        	$(CXX) $(CFLAGS) $(CXXFLAGS) $(CPPFLAGS) -c -o main.o main.cpp
+        """)
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        from conan.tools.gnu import AutotoolsToolchain, Autotools, AutotoolsDeps
+
+        class TestConan(ConanFile):
+            requires = "hello/0.1"
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = "Makefile"
+
+            def generate(self):
+                deps = AutotoolsDeps(self)
+                deps.generate()
+                tc = AutotoolsToolchain(self)
+                tc.generate()
+
+            def build(self):
+                autotools = Autotools(self)
+                autotools.make()
+        """)
+    client.save({"main.cpp": main,
+                 "Makefile": makefile,
+                 "conanfile.py": conanfile,
+                 "profile": profile}, clean_first=True)
+
+    client.run("install . --profile=profile")
+    client.run_command("autotoolsdeps.bat && autotools.bat && {}".format(make_program))
+    print(client.out)
+    client.run_command("app")
+    # TODO: fill compiler version when ready
+    check_exe_run(client.out, "main", "gcc", None, "Release", "x86_64", None)
+    assert "hello/0.1: Hello World Release!" in client.out
+
+    client.save({"main.cpp": gen_function_cpp(name="main", msg="main2",
+                                              includes=["hello"], calls=["hello"])})
+    # Make sure it is newer
+    t = time.time() + 1
+    touch(os.path.join(client.current_folder, "main.cpp"), (t, t))
+
+    client.run("build .")
+    print(client.out)
+    client.run_command("app")
+    # TODO: fill compiler version when ready
+    check_exe_run(client.out, "main2", "gcc", None, "Release", "x86_64", None, cxx11_abi=0)
+    assert "hello/0.1: Hello World Release!" in client.out
+    return client.out
+
+
+@pytest.mark.tool_cygwin
+@pytest.mark.skipif(platform.system() != "Windows", reason="Needs windows")
+def test_autotoolsdeps_cygwin():
+    gcc = textwrap.dedent("""
+        [settings]
+        os=Windows
+        os.subsystem=cygwin
+        compiler=gcc
+        compiler.version=4.9
+        compiler.libcxx=libstdc++
+        arch=x86_64
+        build_type=Release
+        """)
+    out = build_windows_subsystem(gcc, make_program="make")
+    print(out)
+    assert "__MSYS__" not in out
+    assert "MINGW" not in out
+    assert "main2 __CYGWIN__1" in out
+
+
+@pytest.mark.tool_mingw
```

Comment:
```suggestion

@pytest.mark.tool_msys2
@pytest.mark.tool_mingw32
```
not sure about an old `tool_mingw`, let's use the one from the MSYS

Reply:
It is another mingw, quite popular that doesn't work under the umbrella of Msys2. We need to make sure it is working too, the commands, paths, syntax, might be different of what Msys2 is doing, so lets keep this variant too.